subprojects/build-cache-http/src/main/java/org/gradle/caching/http/HttpBuildCache.java

```diff
@@ -85,4 +86,42 @@ public HttpBuildCacheCredentials getCredentials() {
     public void credentials(Action<? super HttpBuildCacheCredentials> configuration) {
         configuration.execute(credentials);
     }
+
+    /**
+     * Specifies whether it is acceptable to communicate with an HTTP build cache backend with an untrusted SSL certificate.
+     * <p>
+     * The SSL certificate for the HTTP build cache backend may be untrusted since it is internally provisioned or a self-signed certificate.
+     * <p>
+     * In such a scenario, you can either configure the build JVM environment to trust the certificate,
+     * or set this property to {@code true} to disable verification of the server's identity.
+     * <p>
+     * Allowing communication with untrusted servers keeps data encrypted during transmission,
+     * but makes it easy for a man-in-the-middle to impersonate the intended server and capture data.
+     * <p>
+     * This value has no effect if a server is specified using the HTTP protocol (i.e. has SSL disabled).
+     *
+     * @since 4.2
+     */
+    public boolean isAllowUntrustedServer() {
+        return allowUntrustedServer;
+    }
+
+    /**
+     * Specifies whether it is acceptable to communicate with an HTTP build cache backend with an untrusted SSL certificate.
+     * <p>
+     * The SSL certificate for the HTTP build cache backend may be untrusted since it is internally provisioned or a self-signed certificate.
+     * <p>
+     * In such a scenario, you can either configure the build JVM environment to trust the certificate,
+     * or set this property to {@code true} to disable verification of the server's identity.
+     * <p>
+     * Allowing communication with untrusted servers keeps data encrypted during transmission,
+     * but makes it easy for a man-in-the-middle to impersonate the intended server and capture data.
```

Comment:
Nit-picking, but "easier" would be more precise.

Reply:
fixed.